From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Wed, 22 Apr 2026 00:26:33 +0200
Subject: backends/color-device: Split out the color device setup

So that we can reuse the common code in an upcoming commit

Part-of: <https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/5038>

Origin: https://gitlab.gnome.org/GNOME/mutter/-/commit/f100142
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/stonking/+source/mutter/+bug/2148655
---
 src/backends/meta-color-device.c | 15 +++++++++++++--
 1 file changed, 13 insertions(+), 2 deletions(-)

diff --git a/src/backends/meta-color-device.c b/src/backends/meta-color-device.c
index 4a22e0e..53fa467 100644
--- a/src/backends/meta-color-device.c
+++ b/src/backends/meta-color-device.c
@@ -94,6 +94,9 @@ G_DEFINE_TYPE (MetaColorDevice, meta_color_device,
 
 static const char *efivar_test_path = NULL;
 
+static void setup_created_cd_device (MetaColorDevice *color_device,
+                                     CdDevice        *cd_device);
+
 /*
  * Generate a colord DeviceId according to
  * `device-and-profiling-naming-spec.txt`.
@@ -502,8 +505,6 @@ on_cd_device_created (GObject      *object,
 {
   CdClient *cd_client = CD_CLIENT (object);
   MetaColorDevice *color_device = user_data;
-  MetaColorManager *color_manager;
-  MetaColorStore *color_store;
   CdDevice *cd_device;
   g_autoptr (GError) error = NULL;
 
@@ -520,6 +521,16 @@ on_cd_device_created (GObject      *object,
       return;
     }
 
+  setup_created_cd_device (color_device, cd_device);
+}
+
+static void
+setup_created_cd_device (MetaColorDevice *color_device,
+                         CdDevice        *cd_device)
+{
+  MetaColorManager *color_manager;
+  MetaColorStore *color_store;
+
   color_device->cd_device = cd_device;
 
   cd_device_connect (cd_device, color_device->cancellable,
